src/js/configHolder.js

```diff
@@ -0,0 +1,153 @@
+            "rootPath": "{that}.options.config.rootPath",
+            "enabled": true,
+            "js": {
+                "minimatchOptions": "{that}.options.config.minimatchOptions",
+                "rootPath": "{that}.options.config.rootPath",
+                "enabled": true,
+                "includes": "{that}.options.config.sources.js",
+                "excludes": [],
+                "options": {
+                    "resolvePluginsRelativeTo": "@expand:fluid.module.resolvePath(%fluid-lint-all)",
+                    "overrideConfig": {}
+                }
+            },
+            "json": {
+                "minimatchOptions": "{that}.options.config.minimatchOptions",
+                "rootPath": "{that}.options.config.rootPath",
+                "enabled": true,
+                "includes": "@expand:fluid.flatten({that}.options.config.sources.json, {that}.options.config.sources.json5)",
+                "excludes": ["./package-lock.json"],
+                options: {
+                    "resolvePluginsRelativeTo": "@expand:fluid.module.resolvePath(%fluid-lint-all)",
+                    "overrideConfig": {
+                        "rules": {
+                            /*
+                                Our approach doesn't work well with leading comments in json5 files, which appear to be incorrectly
+                                indented.  As we check for indentation using lintspaces, we can safely disable that check here.
+                            */
+                            "indent": "off",
+                            /*
+                                Allow ES5 multi-line strings.
+                            */
+                            "no-multi-str": "off",
+                            "trailing-comma": "off"
+                        }
+                    }
+                }
+            },
+            "md": {
+                "minimatchOptions": "{that}.options.config.minimatchOptions",
+                "rootPath": "{that}.options.config.rootPath",
+                "enabled": true,
+                "includes": "{that}.options.config.sources.md",
+                "excludes": [],
+                "options": {
+                    "resolvePluginsRelativeTo": "@expand:fluid.module.resolvePath(%fluid-lint-all)",
+                    "overrideConfig": {
+                        "env": {
+                            "browser": true
+                        },
+                        "rules": {
+                            "no-undef": "off",
+                            "strict": "off",
+                            "no-unused-vars": "off",
+                            "no-console": "off"
+                        },
+                        "plugins": [
+                            "markdown"
+                        ]
+                    }
+                }
+            }
+        },
+        "json5lint": {
+            "minimatchOptions": "{that}.options.config.minimatchOptions",
```

Comment:
Feels like each of these blocks should really actually be components - then they can inherit common options such as minimatchOptions, rootPath, and enabled

Reply:
I made a pass at this.